tests/test_utils.py

```diff
@@ -0,0 +1,358 @@
+        "4294967291",
+        "100",
+        "0",
+        "269877248",
+        "678",
+        "0",
+        "55",
+        "4293968720",
+        "0",
+        "9",
+        "9999",
+    ]
+
+
+@pytest.fixture
+def table_2() -> Dict[str, Iterable]:
+    table: Dict[str, Iterable] = dict(
+        REPEATS=[1, 0],
+        TRIGGER=["Immediate", "Immediate"],
+        POSITION=[-20, 2**31 - 1],
+        TIME1=[12, 2**32 - 1],
+        TIME2=[32, 1],
+    )
+
+    table["OUTA1"] = [False, True]
+    table["OUTA2"] = [True, False]
+    for key in "BCDEF":
+        table[f"OUT{key}1"] = table[f"OUT{key}2"] = [False, False]
+
+    return table
+
+
+@pytest.fixture
+def table_data_2() -> List[str]:
+    return [
+        "67108865",
+        "4294967276",
+        "12",
+        "32",
+        "1048576",
+        "2147483647",
+        "4294967295",
+        "1",
+    ]
+
+
+def test_table_packing_pack_length_mismatched(
+    table_1: OrderedDict[str, Iterable],
+    table_field_info: TableFieldInfo,
+):
+    assert table_field_info.row_words
+
+    # Adjust one of the record lengths so it mismatches
+    field_info = table_field_info.fields[("OUTC1")]
+    assert field_info
+    table_1["OUTC1"] = np.array([1, 2, 3, 4, 5, 6, 7, 8])
+
+    with pytest.raises(AssertionError):
+        table_to_words(table_1, table_field_info)
+
+
+@pytest.mark.parametrize(
+    "table_fixture_name,table_data_fixture_name",
+    [
+        ("table_1_not_in_panda_order", "table_data_1"),
+        ("table_2", "table_data_2"),
+        ("table_1_np_arrays", "table_data_1"),
+    ],
+)
+def test_table_to_words_and_words_to_table(
+    table_fixture_name: str,
+    table_data_fixture_name: str,
+    table_field_info: TableFieldInfo,
+    request,
+):
+    table: Dict[str, Iterable] = request.getfixturevalue(table_fixture_name)
+    table_data: List[str] = request.getfixturevalue(table_data_fixture_name)
+
+    output_data = table_to_words(table, table_field_info)
+    assert output_data == table_data
+    output_table = words_to_table(output_data, table_field_info)
+
+    # Test the correct keys are outputted
+    assert output_table.keys() == table.keys()
+
+    # Check the items have been inserted in panda order
+    sorted_output_table = OrderedDict({key: table[key] for key in output_table.keys()})
+    assert sorted_output_table != OrderedDict(table)
+
+    # Check the values are the same
+    for output_key in output_table.keys():
+        np.testing.assert_equal(output_table[output_key], table[output_key])
+
+
+def test_table_packing_unpack(
+    table_1_np_arrays: OrderedDict[str, np.ndarray],
+    table_field_info: TableFieldInfo,
+    table_data_1: List[str],
+):
+    assert table_field_info.row_words
+    output_table = words_to_table(table_data_1, table_field_info)
+
+    actual: UnpackedArray
+    for field_name, actual in output_table.items():
+        expected = table_1_np_arrays[str(field_name)]
+        np.testing.assert_array_equal(actual, expected)
```

Comment:
Nice find! There is also https://numpy.org/doc/stable/reference/generated/numpy.testing.assert_equal.html so could we `np.testing.assert_equal(output_table, table_1_np_arrays)` instead of this for loop?